hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmLikePredicate.java

```diff
@@ -145,4 +145,10 @@ public void appendHqlString(StringBuilder sb) {
 	protected SqmNegatablePredicate createNegatedNode() {
 		return new SqmLikePredicate( matchExpression, pattern, escapeCharacter, !isNegated(), nodeBuilder() );
 	}
+
+	@Override
+	public SqmNegatablePredicate not() {
+		negate();
+		return this;
+	}
```

Comment:
This is wrong, please revert.

```suggestion
```

Reply:
This was added to solve the failing `ILikeCriteriaTest.testNotIlike`and `ILikeCriteriaTest.testNotIlikeEscape`, due to the fix for the SqmNegatedPredicate